frontend/src/v5/ui/routes/dashboard/projects/containers/createContainerForm/createContainerForm.component.tsx

```diff
@@ -0,0 +1,156 @@
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { formatMessage } from '@/v5/services/intl';
+import { FormattedMessage } from 'react-intl';
+import { TextField, MenuItem } from '@material-ui/core';
+import { FormModal } from '@/v5/ui/controls/modal/formModal/formDialog.component';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { useParams } from 'react-router';
+import { ContainersActionsDispatchers } from '@/v5/services/actionsDispatchers/containersActions.dispatchers';
+import { Select } from './createContainerForm.styles';
+
+interface IFormInput {
+	name: string;
+	unit: string;
+	desc: string;
+	code: string;
+	type: string;
+}
+
+const errorMessage = {
+	name: {
+		required: formatMessage({ id: 'containers.creation.name.error.required', defaultMessage: 'Name is a required field' }),
+		maxLength: formatMessage({ id: 'containers.creation.name.error.length', defaultMessage: 'Name is limited to 120 characters' }),
+	},
+	code: {
+		maxLength: formatMessage({ id: 'containers.creation.code.error.length', defaultMessage: 'Code is limited to 50 characters' }),
+		pattern: formatMessage({ id: 'containers.creation.code.error.pattern', defaultMessage: 'Code is limited to letters and numbers' }),
+	},
+};
+
+export const CreateContainerForm = ({ open, close }): JSX.Element => {
+	const { register, handleSubmit, formState, reset, formState: { errors } } = useForm<IFormInput>({
+		mode: 'onChange',
+	});
+	const { teamspace, project } = useParams() as { teamspace: string, project: string };
+	const onSubmit: SubmitHandler<IFormInput> = (body) => {
+		ContainersActionsDispatchers.createContainer(teamspace, project, body);
+		close();
+	};
+
+	React.useEffect(() => { reset(); }, [!open]);
+
+	return (
+		<FormModal
+			title={formatMessage({ id: 'containers.creation.title', defaultMessage: 'Create new Container' })}
+			open={open}
+			onClickClose={close}
+			onSubmit={handleSubmit(onSubmit)}
+			confirmLabel={formatMessage({ id: 'containers.creation.ok', defaultMessage: 'Create Container' })}
+			isValid={formState.isValid}
+		>
+			<TextField
+				label={formatMessage({ id: 'containers.creation.form.name', defaultMessage: 'Name' })}
+				required
+				error={!!errors.name}
+				helperText={errors.name && errorMessage.name[errors.name?.type]}
+				{...register('name', { required: true, maxLength: 120 })}
```

Comment:
For validation we could use Yup to be consistent all around the system (is being used also in the backend)
https://react-hook-form.com/get-started#schemavalidation